cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -173,18 +170,24 @@ func ValidateDiscoveryFile(discoveryFile string, fldPath *field.Path) field.Erro
 	return allErrs
 }
 
-// ValidateToken validates token
-func ValidateToken(t string, fldPath *field.Path) field.ErrorList {
+func ValidateBootstrapTokens(bts []kubeadm.BootstrapToken, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-
-	id, secret, err := tokenutil.ParseToken(t)
-	if err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath, t, err.Error()))
+	for i, bt := range bts {
+		allErrs = append(allErrs, ValidateToken(bt.Token.String(), fldPath.Child(fmt.Sprintf("%d", i)).Child("token"))...)
+		allErrs = append(allErrs, ValidateTokenUsages(bt.Usages, fldPath.Child(fmt.Sprintf("%d", i)).Child("usages"))...)
+		allErrs = append(allErrs, ValidateTokenGroups(bt.Usages, bt.Groups, fldPath.Child(fmt.Sprintf("%d", i)).Child("groups"))...)
 	}
```

Comment:
I think we should add validation that TTL and Expires are not set at the same time (mutually exclusive). WDYT?

Reply:
I will add that thanks